tests/test_discover_api.py

```diff
@@ -23,6 +26,9 @@ def setup_method(self, test_method):
         self.generate_event()
         self.generate_transaction()
 
+    def teardown_method(self, test_method):
+        self.__dataset_manager.__exit__(None, None, None)
```

Comment:
These weren't being torn down previously — using a context manager fixture instead here is a benefit because teardown essentially comes for free.